### PR TITLE
[Merged by Bors] - feat: port Analysis.NormedSpace.WeakDual

### PR DESCRIPTION
**Pull request by apurvnakade**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/WeakDual.lean [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+map. -/
+def continuousLinearMapToWeakDual : Dual 𝕜 E →L[𝕜] WeakDual 𝕜 E :=
+  { toWeakDual with cont := toWeakDual_continuous }
+#align normed_space.dual.continuous_linear_map_to_weak_dual NormedSpace.Dual.continuousLinearMapToWeakDual
+
+/-- The weak-star topology is coarser than the dual-norm topology. -/
+theorem dual_norm_topology_le_weak_dual_topology :
+    (by infer_instance : TopologicalSpace (Dual 𝕜 E)) ≤
+      (by infer_instance : TopologicalSpace (WeakDual 𝕜 E)) := by
+  convert toWeakDual_continuous.le_induced
```
Comment (apurvnakade) [inline Mathlib/Analysis/NormedSpace/WeakDual.lean:149]: I will definitely not be able to fix this as I don't even understand how there are tactics in the statement of the theorem. Any suggestions on how to deal with this?

Comment (mcdoll) [reply Mathlib/Analysis/NormedSpace/WeakDual.lean:149]: Shouldn't the RHS just be `WeakDual.instTopologicalSpace 𝕜 E`? For the RHS maybe just check what `infer_instance` outputs

Comment (apurvnakade) [reply Mathlib/Analysis/NormedSpace/WeakDual.lean:149]: Here's the goal window with `pp` false:

```lean 
  𝕜 : Type u_1
  inst✝² : NontriviallyNormedField 𝕜
  E : Type u_2
  inst✝¹ : SeminormedAddCommGroup E
  inst✝ : NormedSpace 𝕜 E
  ⊢ ∀ {α β : Type (max ?u.65216 ?u.65217)} {t : TopologicalSpace α} {t' : TopologicalSpace β} {f : α → β},
    Continuous f → t ≤ TopologicalSpace.induced f t'
```

File: Mathlib/Analysis/NormedSpace/WeakDual.lean [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+#align weak_dual.is_closed_polar WeakDual.isClosed_polar
+
+variable {𝕜}
+
+/-- While the coercion `coe_fn : WeakDual 𝕜 E → (E → 𝕜)` is not a closed map, it sends *bounded*
+closed sets to closed sets. -/
+theorem isClosed_image_coe_of_bounded_of_closed {s : Set (WeakDual 𝕜 E)}
+    (hb : Bounded (Dual.toWeakDual ⁻¹' s)) (hc : IsClosed s) :
+    IsClosed ((coeFn : WeakDual 𝕜 E → E → 𝕜) '' s) :=
+  ContinuousLinearMap.isClosed_image_coe_of_bounded_of_weak_closed hb (isClosed_induced_iff'.1 hc)
```
Comment (apurvnakade) [inline Mathlib/Analysis/NormedSpace/WeakDual.lean:217]: Error:

```lean
type mismatch
  ContinuousLinearMap.isClosed_image_coe_of_bounded_of_weak_closed hb (Iff.mp isClosed_induced_iff' hc)
has type
  IsClosed (FunLike.coe '' (↑Dual.toWeakDual ⁻¹' s)) : Prop
but is expected to have type
  IsClosed (coeFn '' s) : Prop
```

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**